lib/features/feed/cubit/share_feed_post_cubit.dart

```diff
@@ -1,30 +1,104 @@
         final files = await Misc.downloadFiles([feed.image!]);
-        await Share.shareXFiles(
-          files,
-          text: '${feed.body}\n\nhttps://fluttercon.dev',
-        );
-      } else {
-        await Share.share('${feed.body}\n\nhttps://fluttercon.dev');
+        if (files.isNotEmpty) {
+          filePath = files.first.path;
+        }
       }
+
+      final installedApps = await appinioSocialShare.getInstalledApps();
+
+      switch (platform) {
+        case Platform.telegram:
+          if (installedApps['telegram'] ?? false) {
+            await _shareToTelegram(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('Telegram is not installed.'));
+          }
+        case Platform.whatsapp:
+          if (installedApps['whatsapp'] ?? false) {
+            await _shareToWhatsApp(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('WhatsApp is not installed.'));
+          }
+        case Platform.twitter:
+          if (installedApps['twitter'] ?? false) {
+            await _shareToTwitter(message, filePath);
+          } else {
+            emit(const ShareFeedPostState.error('Twitter is not installed.'));
+          }
+        case Platform.facebook:
+          if (installedApps['facebook'] ?? false) {
+            if (filePath == null) {
+              emit(
+                const ShareFeedPostState.error(
+                  'Failed to share post. Kindly share a post with an image.',
+                ),
+              );
+            } else {
+              await _shareToFacebook(message, filePath);
+            }
+          } else {
+            emit(const ShareFeedPostState.error('Facebook is not installed.'));
+          }
+      }
+
       emit(const ShareFeedPostState.loaded());
     } catch (e) {
-      emit(ShareFeedPostState.error(e.toString()));
+      emit(ShareFeedPostState.error('Failed to share post: $e'));
+    }
+  }
+
+  Future<void> _shareToTelegram(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToTelegram(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to Telegram: $e');
+    }
+  }
+
+  Future<void> _shareToWhatsApp(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToWhatsapp(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to WhatsApp: $e');
+    }
+  }
+
+  Future<void> _shareToFacebook(String message, String filePath) async {
+    try {
+      await appinioSocialShare.android.shareToFacebook(message, [filePath]);
+    } catch (e) {
+      throw Exception('Error sharing to Facebook: $e');
+    }
+  }
+
+  Future<void> _shareToTwitter(String message, String? filePath) async {
+    try {
+      await appinioSocialShare.android.shareToTwitter(message, filePath);
+    } catch (e) {
+      throw Exception('Error sharing to Twitter: $e');
     }
   }
```

Comment:
Please append the Fluttercon website to the message, like so `\n\nhttps://fluttercon.dev'` as it was prior

Reply:
Scratch [this](https://github.com/droidconKE/flutterconKEApp/pull/91#discussion_r1720326983). Seen what you did.